src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -1445,6 +1460,11 @@ internal bool ReportExplicitUseOfReservedAttributes(in DecodeWellKnownAttributeA
                 // Do not use 'System.Runtime.CompilerServices.RequiredMemberAttribute'. Use the 'required' keyword on required fields and properties instead.
                 diagnostics.Add(ErrorCode.ERR_ExplicitRequiredMember, arguments.AttributeSyntaxOpt.Location);
             }
+            else if (attribute.IsTargetAttribute(this, AttributeDescription.CompilerFeatureRequiredAttribute))
+            {
+                // This is always disallowed, so we don't bother with a flag.
```

Comment:
>// This is always disallowed, so we don't bother with a flag. [](http://example.com/codeflow?start=16&length=61)

How confident are we that a call to this method won't be missed because there is no specific flag value to pass?

Reply:
> How confident are we that a call to this method won't be missed because there is no specific flag value to pass?

I'm not certain how a flag would improve that scenario. I already had to fix a bug that this method wasn't called for one scenario that _did_ have a flag.

Reply:
> I'm not certain how a flag would improve that scenario. I already had to fix a bug that this method wasn't called for one scenario that did have a flag.

Exactly my point. This is just a convenience helper. No one is responsible to call it unless they want to. As opposed to ```DecodeWellKnownAttribute```, which is getting always called by the pipeline. The change in SourceTypeParameterSymbol looks more like a refactoring rather than a bug fix to me. 

I suggest to take approach that is used in VB. Put this check into 
```
internal virtual void DecodeWellKnownAttribute(ref DecodeWellKnownAttributeArguments<AttributeSyntax, CSharpAttributeData, AttributeLocation> arguments)
```

Ensure overrides call into the base, we do have a bug around that.
    
Regarding the flag. At least the flag would be in the list for one to find as something to prevent. But I agree, that it is not a 100% protection.